### PR TITLE
fix: `AttributeError` when comparing commands

### PR DESCRIPTION
**Pull request by Dorukyum**
## Summary
Fixes `AttributeError`s that occur when comparing application commands with non-command objects.

## Information

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

**Comment by Lulalaby** [comment]
mhh i would be careful with such a check.
a simple dict eq check won't do i fear and might cause ratelimit hits.

**Comment by Dorukyum** [comment]
@Lulalaby I don't quite understand why this won't do, what's the issue and what do you suggest?

**Comment by Lulalaby** [comment]
discord tends to mess up the received orders of properties and send back complete different ones.
I'm not sure how pythons methods work but for c# it caused many problems.
I feel like this wouldn't make it precise.. instead more unprecise
I might be wrong

**Comment by Dorukyum** [comment]
I might've overlooked stuff like that, will look into our sync algorithm again to see if this could cause any issues.

**Comment by plun1331** [comment]
> discord tends to mess up the received orders of properties and send back complete different ones. I'm not sure how pythons methods work but for c# it caused many problems. I feel like this wouldn't make it precise.. instead more unprecise I might be wrong

if you're referring to the order in which the data is sent back, i believe dictionaries are unordered by default

**Comment by Dorukyum** [comment]
Yes, that's right, but I figured I'd still check everything in case some info is missing somewhere or whatever (`{"a": 1}` is not equal to `{"a": 1, "b": 2}`). Will update today.

**Comment by Dorukyum** [comment]
I think the only thing this will break is `Bot.add_application_command` as it loops through pending application commands to update their IDs. I suggest we update that method before merging this PR to fix the `AttributeError`s, but eventually refactor the way commands are stored.

**Comment by Lulalaby** [comment]
_good that i reminded you to check_

**Comment by Dorukyum** [comment]
Now that I think about it, it actually comes down to what we want this method to check: either the commands having the same frontend (the same name in the same guilds) or the commands being completely identical. I reworked this pull request to only fix the potential `AttributeError`s, while keeping the original functionality of the method.

File: discord/commands/core.py [hunk 1]
```diff
@@ -235,15 +235,10 @@ def __repr__(self) -> str:
...
-            and getattr(other, "id", None) is not None
-        ):
-            check = self.id == other.id
-        else:
-            check = self.name == other.name and self.guild_ids == other.guild_ids
         return (
-            isinstance(other, self.__class__) and self.parent == other.parent and check
+            isinstance(other, self.__class__)
+            and self.qualified_name == other.qualified_name
+            and self.guild_ids == other.guild_ids
```
Comment (Lulalaby) [inline discord/commands/core.py:241]: can this throw?

Comment (Dorukyum) [reply discord/commands/core.py:241]: No, this won't throw any AttributeErrors. The isinstance check makes sure of that.

**Review by Lulalaby — DISMISSED** [review]

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**